test/stmatic-test.ts

```diff
@@ -0,0 +1,1074 @@
+
+            expect(balanceAfter.lt(ethers.utils.parseEther(withdrawAmounts[i])))
+                .to.be.true;
+        }
+    });
+
+    it("Requesting withdraw BEFORE slashing should result in a lower balance withdrawal", async () => {
+        const ownedTokens: BigNumber[][] = [];
+        const submitAmounts: string[] = [];
+        const withdrawAmounts: string[] = [];
+
+        const [minAmount, maxAmount] = [0.001, 0.1];
+        const delegatorsAmount = Math.floor(Math.random() * (10 - 1)) + 1;
+        const testersAmount = Math.floor(Math.random() * (10 - 1)) + 1;
+        for (let i = 0; i < delegatorsAmount; i++) {
+            await mint(testers[i], ethers.utils.parseEther("100"));
+
+            await addOperator(
+                `BananaOperator${i}`,
+                testers[i].address,
+                ethers.utils.randomBytes(64)
+            );
+
+            await stakeOperator(i + 1, testers[i], "10");
+        }
+
+        for (let i = 0; i < testersAmount; i++) {
+            submitAmounts.push(
+                (
+                    (Math.random() * (maxAmount - minAmount) + minAmount) *
+                    delegatorsAmount
+                ).toFixed(3)
+            );
+            const submitAmountWei = ethers.utils.parseEther(submitAmounts[i]);
+
+            await mint(testers[i], submitAmountWei);
+            await submit(testers[i], submitAmountWei);
+        }
+
+        await stMATIC.delegate();
+
+        for (let i = 0; i < testersAmount; i++) {
+            withdrawAmounts.push(
+                (
+                    Math.random() * (Number(submitAmounts[i]) - minAmount) +
+                    minAmount
+                ).toFixed(3)
+            );
+            const withdrawAmountWei = ethers.utils.parseEther(
+                withdrawAmounts[i]
+            );
+            await requestWithdraw(testers[i], withdrawAmountWei);
+            ownedTokens.push(
+                await poLidoNFT.getOwnedTokens(testers[i].address)
+            );
+        }
+
+        for (let i = 0; i < delegatorsAmount; i++) {
+            await slash(i + 1, 10);
+        }
+
+        const withdrawalDelay = await mockStakeManager.withdrawalDelay();
+        const currentEpoch = await mockStakeManager.epoch();
+        await mockStakeManager.setEpoch(withdrawalDelay.add(currentEpoch));
+
+        for (let i = 0; i < testersAmount; i++) {
+            for (let j = 0; j < ownedTokens[i].length; j++) {
+                await claimTokens(testers[i], ownedTokens[i][j]);
+            }
+            const balanceAfter = await mockERC20.balanceOf(testers[i].address);
+
+            expect(balanceAfter.lt(ethers.utils.parseEther(withdrawAmounts[i])))
+                .to.be.true;
+        }
+    });
+
+    it("Should pause the contract successfully", async () => {
+        await stMATIC.togglePause();
+        await expect(stMATIC.delegate()).to.be.revertedWith("Pausable: paused");
+    });
+
+    // todo test with totalBuffred != 0
```

Comment:
Is this TODO done?

Reply:
Yes it's done
https://github.com/Shard-Labs/PoLido/blob/pre-testnet-fixes/test/stmatic-test.ts#L704

Reply:
@GitMark0 Then remove this TODO comment please.